fmt/src/formatter.rs

```diff
@@ -152,10 +151,18 @@ impl<'a, W: Write> Formatter<'a, W> {
 
     /// Get the Write interface of the current temp buffer or the underlying Write
     fn buf(&mut self) -> &mut dyn Write {
-        if self.temp_bufs.is_empty() {
-            &mut self.buf as &mut dyn Write
-        } else {
-            self.temp_bufs.last_mut().unwrap() as &mut dyn Write
+        match &mut self.temp_bufs[..] {
+            [] => &mut self.buf as &mut dyn Write,
+            [.., buf] => buf as &mut dyn Write,
+        }
+    }
+
+    /// Casts the current `W` writer as a `String` reference. Should only be used for debugging.
+    #[allow(dead_code)]
+    unsafe fn buf_contents(&self) -> &String {
+        match &self.temp_bufs[..] {
+            [] => unsafe { *(&self.buf.w as *const W as *const &mut String) },
+            [.., buf] => &buf.w,
         }
     }
```

Comment:
Used inline with `trace!("{}", self.buf_contents())`

This avoids adding `fmt::Debug` bounds everywhere, makes debugging faster. LMK if this should be removed.